onnxruntime/core/providers/qnn/builder/opbuilder/slice_op_builder.cc

```diff
@@ -84,123 +123,68 @@ Status SliceOpBuilder::ProcessInputs(QnnModelWrapper& qnn_model_wrapper,
   if (do_op_validation) {
     ORT_RETURN_IF_ERROR(ExplictOpCheck(qnn_model_wrapper, node_unit));
   }
-  Qnn_DataType_t qnn_data_type = QNN_DATATYPE_FLOAT_32;
+
+  // Only need to add input 0. The other inputs (if any) contain static data that is passed to QNN APIs
+  // as static parameters.
+  return ProcessInput(qnn_model_wrapper, node_unit.Inputs()[0], logger, input_names);
+}
+
+Status SliceOpBuilder::ProcessAttributesAndOutputs(QnnModelWrapper& qnn_model_wrapper,
+                                                   const NodeUnit& node_unit,
+                                                   std::vector<std::string>&& input_names,
+                                                   const logging::Logger& logger,
+                                                   bool do_op_validation) const {
+  // Extract starts, ends, axes, and steps data from attributes (opset < 10) or initializer inputs (opset >= 10).
   TensorShapeVector raw_starts;
   TensorShapeVector raw_ends;
   TensorShapeVector raw_axes;
   TensorShapeVector raw_steps;
-  std::vector<uint32_t> input0_shape;
 
-  auto inputs = node_unit.Inputs();
-  auto input_count = inputs.size();
-  // Opset 9, only 1 input, starts, ends, axes are in attribute
-  if (1 == input_count) {
+  const auto& inputs = node_unit.Inputs();
+  const size_t input_count = inputs.size();
+
+  // Opset 9 only has 1 input. The starts, ends, axes values are attributes.
+  if (node_unit.SinceVersion() < 10) {
     GetDataFromAttribute(node_unit, raw_starts, raw_ends, raw_axes);
-  }
+  } else {
+    constexpr size_t starts_index = 1;
+    constexpr size_t ends_index = 2;
+    constexpr size_t axes_index = 3;
+    constexpr size_t steps_index = 4;
 
-  for (size_t input_i = 0; input_i < input_count; ++input_i) {
-    auto& input_name = inputs[input_i].node_arg.Name();
-    if (input_name.empty()) {
-      // Ignore unspecified/unused optional input
-      continue;
-    }
-    if (qnn_model_wrapper.IsQnnTensorWrapperExist(input_name)) {
```

Comment:
Note: This early check caused the loop to continue and prevented the `raw_starts` or `raw_ends` data structures from being properly initialized. They should always be initialized even if another operator has already added them to the QNN model.